File: apps/android/core/model/src/main/kotlin/com/sigmundgranaas/turbo/expressive/domain/Ntb.kt
package com.sigmundgranaas.turbo.expressive.domain

/** What kind of Nasjonal Turbase (ut.no / DNT) object a marker represents. */
enum class NtbPoiType { Cabin, Trip, Place }

/**
 * A lightweight Nasjonal Turbase marker (cabin / trip / place) as served by the
 * backend proxy (`/api/places/ntb`). The proxy already normalised it, so this is
 * a thin value type. A [Trip] additionally has a route polyline fetched lazily
 * via [NasjonalTurbaseRepository.route].
 */
data class NtbPoi(
    val id: String,
    val type: NtbPoiType,
    val title: String,
    val position: LatLng,
    val summary: String? = null,
    val imageUrl: String? = null,
    /** Best link back to ut.no (proxy-provided), or null. */
    val utUrl: String? = null,
) {
    val hasRoute: Boolean get() = type == NtbPoiType.Trip
}

/** A trip's full detail: the route polyline to reveal plus sheet metadata. */
data class NtbRoute(
    val id: String,
    val title: String,
    val points: List<LatLng>,
    val description: String? = null,
    val distanceMeters: Double? = null,
    val grade: String? = null,
    val imageUrl: String? = null,
    val utUrl: String? = null,
) {
    val hasGeometry: Boolean get() = points.size >= 2
}

File: apps/ios/Sources/CoreModel/Ntb.swift
import Foundation

/// What kind of Nasjonal Turbase (ut.no / DNT) object a marker represents.
/// Mirrors `domain.NtbPoiType` (Android).
public enum NtbPoiType: String, Sendable, Codable {
    case cabin
    case trip
    case place
}

/// A lightweight Nasjonal Turbase marker (cabin / trip / place) as served by the
/// backend proxy (`/api/places/ntb/pois`). The proxy already normalised it, so
/// this is a thin value type. A `.trip` additionally has a route polyline fetched
/// lazily via ``NasjonalTurbaseRepository/route(id:)``. Mirrors `domain.NtbPoi`.
public struct NtbPoi: Identifiable, Hashable, Sendable, Codable {
    public let id: String
    public let type: NtbPoiType
    public let title: String
    public let position: LatLng
    public let summary: String?
    public let imageUrl: String?
    /// Best link back to ut.no (proxy-provided), or `nil`.
    public let utUrl: String?

    public init(
        id: String,
        type: NtbPoiType,
        title: String,
        position: LatLng,
        summary: String? = nil,
        imageUrl: String? = nil,
        utUrl: String? = nil
    ) {
        self.id = id
        self.type = type
        self.title = title
        self.position = position
        self.summary = summary
        self.imageUrl = imageUrl
        self.utUrl = utUrl
    }

    public var hasRoute: Bool { type == .trip }
}

/// A trip's full detail: the route polyline to reveal plus sheet metadata.
/// Mirrors `domain.NtbRoute`.
public struct NtbRoute: Identifiable, Hashable, Sendable, Codable {
    public let id: String
    public let title: String
    public let points: [LatLng]
    public let description: String?
    public let distanceMeters: Double?
    public let grade: String?
    public let imageUrl: String?
    public let utUrl: String?

    public init(
        id: String,
        title: String,
        points: [LatLng],
        description: String? = nil,
        distanceMeters: Double? = nil,
        grade: String? = nil,
        imageUrl: String? = nil,
        utUrl: String? = nil
    ) {
        self.id = id
        self.title = title
        self.points = points
        self.description = description
        self.distanceMeters = distanceMeters
        self.grade = grade
        self.imageUrl = imageUrl
        self.utUrl = utUrl
    }

    public var hasGeometry: Bool { points.count >= 2 }
}
